plugins/flytekit-spark/flytekitplugins/spark/sd_transformers.py

```diff
@@ -56,11 +56,12 @@ def decode(
         flyte_value: literals.StructuredDataset,
         current_task_metadata: StructuredDatasetMetadata,
     ) -> DataFrame:
-        user_ctx = FlyteContext.current_context().user_space_params
+        # The Spark session already exists at this stage - this command will get it
+        spark = SparkSession.builder.getOrCreate()
         if current_task_metadata.structured_dataset_type and current_task_metadata.structured_dataset_type.columns:
             columns = [c.name for c in current_task_metadata.structured_dataset_type.columns]
-            return user_ctx.spark_session.read.parquet(flyte_value.uri).select(*columns)
```

Comment:
you can't get spark session from `user_ctx.spark_session`?

Reply:
how does your task look like? is possible for you to share the code

Reply:
are you using spark decoder in the regular python task or spark task?

Reply:
Thanks for having a look at my changes.



> you can't get spark session from `user_ctx.spark_session`?
No, I can't.  I'll show you why this is the case below.


> how does your task look like? is possible for you to share the code
```python
from __future__ import annotations
from flytekitplugins.spark import Spark, new_spark_session
from flytekit import task, StructuredDataset
from pyspark.sql import SparkSession, DataFrame

@task(task_config=Spark(...))
def demo_spark_task(dataset: StructuredDataset, spark_confs: dict[str, str]) -> None:
    spark = new_spark_session("flyte", conf=spark_confs)
    dataframe = dataset.open(DataFrame).all() # This fails due to user_ctx pointing to old spark session
    ...
```
> are you using spark decoder in the regular python task or spark task?
I am using the spark decoder in a spark task.

Reply:
What I'm trying to achieve here is that I want to set spark confs at execution time. I've seen a PR from July where there was a `.with_overrides(...)` method attached implemented for workflows.

Even though I still think this PR is still a sensible addition to the flytekit codebase, can I configure a Spark task at execution time with the override method?
https://github.com/flyteorg/flytekit/commit/4b1675ffb85648dc5742e9a6dea98b94714963e1

Reply:
why do you need to create another spark `session` in the demo_spark_task? if you don't create it, decoder should work right?

Reply:
I run this code locally, and it works for me
```python
def test_structured_dataset():
    @task(task_config=Spark())
    def my_dataset() -> StructuredDataset:
        session = flytekit.current_context().spark_session
        df = session.createDataFrame([("Megan", 2.0), ("Wayne", float("nan")), ("Dennis", 8.0)], ["name", "age"])
        return df

    @task(task_config=Spark())
    def print_dataset(sd: StructuredDataset):
        print(sd.open(pyspark.sql.DataFrame))

    @workflow
    def my_wf():
        print_dataset(sd=my_dataset())
```